src/sos_polynomial_in_semialgebraic_set_bridge.jl

```diff
@@ -9,10 +9,9 @@ function lagrangian_multiplier(model::MOI.ModelLike, p, set::SOSLikeCone, q,
     # If maxdegree_s2 is odd, div(maxdegree_s2,2) would make s^2 have degree up to maxdegree_s2-1
     # for this reason, we take div(maxdegree_s2+1,2) so that s^2 have degree up to maxdegree_s2+1
     maxdegree_s = div(maxdegree_s2 + 1, 2)
-    # FIXME handle the case where `p`, `q_i`, ...  do not have the same variables
-    # so instead of `variable(p)` we would have the union of them all
-    @assert variables(q) ⊆ variables(p)
-    monos = MP.monomials(MP.variables(p), mindegree_s:maxdegree_s)
+    vars = [MP.variables(p)..., MP.variables(q)...]
+    unique!(vars)
+    monos = MP.monomials(sort(vars, rev=true), 0:maxdegree_s)
```

Comment:
Why use `0:maxdegree_s` instead of `mindegree_s:maxdegree_s` ?

Reply:
This is probably what makes the mock tests fail

Reply:
Maybe I did not understand what mindegree in the Input is. I assumed it to be the minimal degree of p.  But this doesn't make sense. Say p is homogeneous of degree 4 and q is quadratic. Then mindegree(p) = maxdegree(p) = 4 and mindegree(q)=0, maxdegree(q)=2. Consequently, mindegree_s2 = 4 and maxdegree_s2 = 2, which is definititly wrong...

Reply:
Indeed, we should probably use 0 here do you have a good example to add as a test ?

Reply:
Here is an example that shows that 0:maxdegree_s is necessary:

    @polyvar x y
    m = SOSModel(factory)
    p = x^4 - y^4 
    S = @set - x^2 - y^4 >=0
    soscon = @constraint m p >= 0 domain = S 
    JuMP.optimize!(m)

One possible sos_decomposition is
(x)^2 + (x^2)^2  + (1)^2 * (-y^4 - x^2)
However, mindegree_s is 1 in this example and indeed, on master the SDP is infeasible.